"display" property value of computed style for audio element without controls is "inline" (not "none")

Ahmad-S792 opened the pull request:
#### 13c796fb7f1d6bc0cb660bedae2f9f6e8e510675
<pre>
&quot;display&quot; property value of computed style for audio element without controls is &quot;inline&quot; (not &quot;none&quot;)

&quot;display&quot; property value of computed style for audio element without controls is &quot;inline&quot; (not &quot;none&quot;)

<a href="https://bugs.webkit.org/show_bug.cgi?id=88105">https://bugs.webkit.org/show_bug.cgi?id=88105</a>

Reviewed by Devin Rousso.

This is to align Webkit behavior with Blink / Chrome and Gecko / Firefox and Web-Specifications:

<a href="https://html.spec.whatwg.org/#embedded-content-rendering-rules">https://html.spec.whatwg.org/#embedded-content-rendering-rules</a>

&quot;When an audio element is not exposing a user interface, the user agent is expected
to force its &apos;display&apos; property to compute to &apos;none&apos;, irrespective of CSS rules.&quot;

When an audio element has no controls attribute, it should not be visible on the page.
Before this patch, when we removed the &quot;controls&quot; attribute, an empty div of the size of the controls remained.

Tests from Patch Authored by Silvia Pfeiffer.

* Source/WebCore/css/html.css: Update UA stylesheet where the audio controls are not visible
* LayoutTests/media/audio-controls-hidden-display-none.html: Added Test Cases
* LayoutTests/media/audio-controls-hidden-display-none-expected.txt: Added Test Case Expectations

Canonical link: <a href="https://commits.webkit.org/255528@main">https://commits.webkit.org/255528@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7b506ffe156add0a515912c02e2ae37fd9a7a144

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92755 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1968 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23338 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102486 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96757 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1969 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30315 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85159 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98628 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98418 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/1333 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79244 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28264 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/83247 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82958 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/71349 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36719 "Built successfully") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16876 "Found 1 new test failure: webanimations/accelerated-web-animation-with-single-interval-and-easing-y-axis-above-1.html (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34516 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18064 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3830 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38387 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40667 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37232 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebCore/css/html.css:
```diff
@@ -55,6 +55,10 @@ address, article, aside, div, footer, header, hgroup, main, nav, section {
     display: block;
 }
 
+audio:not([controls]) {
```
please move this down to be in the `/* media elements */` "section", ideally right below the existing `audio` ruleset

Ahmad-S792 replied:
Done

Ahmad-S792 committed:
"display" property value of computed style for audio element without controls is "inline" (not "none")

"display" property value of computed style for audio element without controls is "inline" (not "none")

https://bugs.webkit.org/show_bug.cgi?id=88105

Reviewed by Devin Rousso.

This is to align Webkit behavior with Blink / Chrome and Gecko / Firefox and Web-Specifications:

https://html.spec.whatwg.org/#embedded-content-rendering-rules

"When an audio element is not exposing a user interface, the user agent is expected
to force its 'display' property to compute to 'none', irrespective of CSS rules."

When an audio element has no controls attribute, it should not be visible on the page.
Before this patch, when we removed the "controls" attribute, an empty div of the size of the controls remained.

Tests from Patch Authored by Silvia Pfeiffer.

* Source/WebCore/css/html.css: Update UA stylesheet where the audio controls are not visible
* LayoutTests/media/audio-controls-hidden-display-none.html: Added Test Cases
* LayoutTests/media/audio-controls-hidden-display-none-expected.txt: Added Test Case Expectations

Canonical link: https://commits.webkit.org/255528@main
